optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py

```diff
@@ -0,0 +1,717 @@
+
+            # 3. Encode input prompt
+            text_encoder_lora_scale = (
+                self.cross_attention_kwargs.get("scale", None) if self.cross_attention_kwargs is not None else None
+            )
+            prompt_embeds, negative_prompt_embeds = self.encode_prompt(
+                prompt,
+                device,
+                num_images_per_prompt,
+                self.do_classifier_free_guidance,
+                negative_prompt,
+                prompt_embeds=prompt_embeds,
+                negative_prompt_embeds=negative_prompt_embeds,
+                lora_scale=text_encoder_lora_scale,
+                clip_skip=self.clip_skip,
+            )
+
+            if ip_adapter_image is not None:
+                image_embeds = self.prepare_ip_adapter_image_embeds(
+                    ip_adapter_image, device, batch_size * num_images_per_prompt
+                )
+
+            # 4. Preprocess image
+            image = self.image_processor.preprocess(image)
+
+            # 5. set timesteps
+            timesteps, num_inference_steps = retrieve_timesteps(self.scheduler, num_inference_steps, device, timesteps)
+            latent_timestep = timesteps[:1].repeat(batch_size * num_images_per_prompt)
+
+            # 6. Prepare latent variables
+            latents = self.prepare_latents(
+                image,
+                latent_timestep,
+                batch_size,
+                num_images_per_prompt,
+                prompt_embeds.dtype,
+                device,
+                generator,
+            )
+
+            # 7. Prepare extra step kwargs. TODO: Logic should ideally just be moved out of the pipeline
+            extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
+
+            # 7.1 Add image embeds for IP-Adapter
+            added_cond_kwargs = {"image_embeds": image_embeds} if ip_adapter_image is not None else None
+
+            # 7.2 Optionally get Guidance Scale Embedding
+            timestep_cond = None
+            if self.unet.config.time_cond_proj_dim is not None:
+                guidance_scale_tensor = torch.tensor(self.guidance_scale - 1).repeat(
+                    batch_size * num_images_per_prompt
+                )
+                timestep_cond = self.get_guidance_scale_embedding(
+                    guidance_scale_tensor, embedding_dim=self.unet.config.time_cond_proj_dim
+                ).to(device=device, dtype=latents.dtype)
+
+            # 8. Split into batches (HPU-specific step)
+            latents_batches, text_embeddings_batches, num_dummy_samples = self._split_inputs_into_batches(
+                batch_size,
+                latents,
+                prompt_embeds,
+                negative_prompt_embeds,
+            )
+
+            outputs = {
+                "images": [],
+                "has_nsfw_concept": [],
+            }
+
+            t0 = time.time()
+            t1 = t0
+
+            hb_profiler = HabanaProfile(
```

Comment:
Please verify profiling options:
```
python image_to_image_generation.py \
    --model_name_or_path "ogkalu/Comic-Diffusion"\
    --src_image_path "
[https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg"](https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg%22)
\
    --prompts "turn him into cyborg" \
    --num_images_per_prompt 20 \
    --profiling_warmup_steps 3 \
    --profiling_steps 2 \
    --batch_size 2 \
    --guidance_scale 7.5 \
    --image_guidance_scale 1 \
    --num_inference_steps 10 \
    --image_save_dir stable_diffusion_images \
    --use_habana \
    --use_hpu_graphs \
    --gaudi_config Habana/stable-diffusion \
    --bf16
```
Error
```
Traceback (most recent call last):
  File "/root/PR/pi314ever_oh/examples/stable-diffusion/image_to_image_generation.py", line 338, in <module>
    main()
  File "/root/PR/pi314ever_oh/examples/stable-diffusion/image_to_image_generation.py", line 300, in main
    outputs = pipeline(
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py", line 560, in __call__
    hb_profiler.step()
  File "/usr/local/lib/python3.10/dist-packages/torch/profiler/profiler.py", line 727, in step
    self._transit_action(prev_action, self.current_action)
  File "/usr/local/lib/python3.10/dist-packages/torch/profiler/profiler.py", line 744, in _transit_action
    action()
  File "/usr/local/lib/python3.10/dist-packages/habana_frameworks/torch/activity_profiler/activity_profiler.py", line 82, in start_trace
    super().start_trace()
  File "/usr/local/lib/python3.10/dist-packages/torch/profiler/profiler.py", line 160, in start_trace
    assert self.profiler is not None
AssertionError
```

Reply:
I will look into this tomorrow, as I am less familiar with how the profiler works.

Reply:
Fixed, just missed a `hb_profiler.start()`